### DISCUSSION TITLE
How to add new operator to onnx?

### DISCUSSION THREAD

**Discussion by ArtyZe** [onnx/onnx#2920] [Q&A]
I have successfully add new operator to torch/onnx, and then I need to do some optimizer for my onnx model. But onnx lib which installed through pip don't recognize my new op, so check failed.
Could you give me some doc about how to add new op to onnx lib? :(
Thanks

**Comment by manhongnie** [comment]
@ArtyZe   Can you tell me how you succeeded? I'm curious, I tried before but failed. Add a new node on onnx

**Comment by ArtyZe** [comment]
> @ArtyZe Can you tell me how you succeeded? I'm curious, I tried before but failed. Add a new node on onnx

Not success add op to onnx and onnxruntime, but success add op to torch/onnx, so I can convert my torch model with new op to onnx successfully, and failed optimize onnx model by onnx and test by onnxruntime

**Comment by manhongnie** [comment]
It may indicate that your node has not been added successfully, or that your operation of the node is wrong.

**Comment by ArtyZe** [comment]
> It may indicate that your node has not been added successfully, or that your operation of the node is wrong.

That means onnx lib will call torch functions? I think onnx functions like checker will call functions in this file before
![image](https://user-images.githubusercontent.com/32027268/88376612-57506680-cdd0-11ea-9789-7d4131828025.png)
And.............what means that "my node is wrong" ?

**Comment by jcwchen** [comment]
Hi @ArtyZe,
https://github.com/onnx/onnx/blob/master/docs/AddNewOp.md can help. Thanks.

**Comment by faxu** [comment]
@ArtyZe if you are using ONNX Runtime to run the model, this can be supported via custom ops. See https://github.com/microsoft/onnxruntime/blob/master/docs/AddingCustomOp.md If you have questions on that, you can post in the [ONNX Runtime github discussions page](https://github.com/microsoft/onnxruntime/discussions)

This will be much easier/quicker than proposing the op to be added to the official ONNX standard.
